### ISSUE TITLE
New Ontology Request: Ontology for Avida digital evolution platform (OntoAvida)

### ISSUE THREAD

**Issue by miguelfortuna** [OBOFoundry/OBOFoundry.github.io#1581]
Use this form to register a new ontology with the OBO Foundry. Please read the instructions provided here:
http://obofoundry.org/docs/NewOntologyRegistrationInstructions.html

## Ontology title
OntoAvida: ontology for Avida digital evolution platform

## Requested ID space
ONTOAVIDA

## Ontology location
https://gitlab.com/fortunalab/ontoavida
latest version in the "development" branch (soon to be released) 

## Contact person
Name: Miguel A. Fortuna
Email address: fortuna@ebd.csic.es
GitHub username: miguelfortuna

## Issue tracker
https://gitlab.com/fortunalab/ontoavida/-/issues

## Ontology license
- [ ] CC0 (public domain)
- [X] CC-BY (version 3 or later)
- [ ] Other: *please specify*

## Available ontology formats
OWL
OBO

## What domain is the ontology intended to cover?
OntoAvida covers the most widely-used computational approach for studying evolution using digital organisms (i.e., self-replicating computer programs that evolve within a user-defined computational environment).

## Related OBO Foundry ontologies
OGG

## Intended use/related projects
We are building AvidaDB, an RDF database on digital organisms (genome sequences, encoded phenotypes, ...), and an R package (avidaR) to facilitate data retrieval to the R community.

## Data source
https://gitlab.com/fortunalab/ontoavida/-/blob/development/ontoavida.owl

## Additional comments or remarks
We are writing a paper and it would be great if we can already mention that the ontology is already part of the OBO Foundry list of ontologies. 

## Metadata

Please fill in the following metadata record which will be used by the OBO Foundry website. Note that the values shown are just examples, for example `yourfourletterid` could be something like `aism`, `cohm`, `mondo` (it does not have to be four letters). `your_domain_like_for_example_anatomy` could be simply `anatomy`, and the license should be whatever your actual license is. An example can be found [here](https://github.com/OBOFoundry/OBOFoundry.github.io/edit/master/ontology/amphx.md), but you really only need to fill in the metadata mentioned here.

```
id: ontoavida
title: OntoAvida: ontology for Avida digital evolution platform
contact:
  email: fortuna@ebd.csic.es
  label: Miguel A. Fortuna
description: OntoAvida develops an integrated vocabulary for the description of the most widely-used computational approach for studying evolution using digital organisms (i.e., self-replicating computer programs that evolve within a user-defined computational environment).
domain: digital evolution, a branch of Artificial Life.
homepage: https://gitlab.com/fortunalab/ontoavida
products:
  - id: ontoavida.owl
  - id: ontoavida.obo
dependencies:
  - id: fbcv
  - id: gsso
  - id: ogg
  - id: stato
  - id: ro
  - id: ncit
tracker: https://gitlab.com/fortunalab/ontoavida/-/issues
license:
  url: http://creativecommons.org/licenses/by/4.0/
  label: CC-BY
usages:
  - user: http://website.of.the.project.using.my.ontology.com
    description: A short description of how the project above is using my ontology
```

**Comment by matentzn** [comment]
@miguelfortuna Thank you very much for your submission! Before we start the review process, we would like to ask you to 
- check a few boxes in the following checklist, and
- ensure that OntoAvida passes the general OBO quality control (OBO Dashboard)

### Submission Checklist

To be considered for inclusion in the OBO Foundry, an ontology must meet certain requirements, as described in [the registration process instructions](http://obofoundry.org/docs/Policy_for_OBO_namespace_and_associated_PURL_requests.html) and the [registration review checklist](https://github.com/OBOFoundry/OBOFoundry.github.io/blob/master/docs/RegistrationChecklist.md). To ensure you are aware of some of its key points, please review the checklist below.

You can either check a box by submitted the request first and then using the GitHub interface, or replacing the `- [ ]` by `- [X]` in the following.

- [ ] I have read an understood [the registration process instructions](http://obofoundry.org/docs/Policy_for_OBO_namespace_and_associated_PURL_requests.html) and the [registration checklist](https://github.com/OBOFoundry/OBOFoundry.github.io/blob/master/docs/RegistrationChecklist.md) 
- [ ] There is no other ontology in the OBO Foundry which would be an appropriate place for my terms. If there was, I have contacted the editors, and we decided in mutual agreement that a separate ontology is more appropriate.
- [ ] My ontology has a specific release file with a version IRI and a dc:license annotation, serialised in RDF/XML.
- [ ] I understand that term definitions, while not mandatory, are key to understanding the intentions of a term especially when the ontology is used in curation. I made sure that a reasonable majority of terms in my ontology have definitions, using the [IAO:0000115](http://www.ontobee.org/ontology/IAO?iri=http://purl.obolibrary.org/obo/IAO_0000115) property.
- [ ] For every term in my ontology, I checked whether another OBO Foundry ontology has one with the same meaning. If so, I re-used that term directly (not by cross-reference, by directly using the IRI).
- [ ] For all relationship properties (Object and Data Property) I checked whether [RO](http://www.ontobee.org/ontology/catalog/RO?iri=http://www.w3.org/2002/07/owl%23ObjectProperty) includes an appropriate one. I understand that aligning with RO is an essential part of the overall alignment between OBO ontologies!
- [ ] For the selection of appropriate annotation properties, I looked at [OMO](http://www.ontobee.org/ontology/catalog/OMO?iri=http://www.w3.org/2002/07/owl%23AnnotationProperty) first. I understand that aligning ontology metadata and term-level metadata is essential for cross-integration of OBO ontologies. 
- [ ] If I was not sure about the meaning of any of the checkboxes above, I have consulted with a member of the OBO Foundry for advice.

### OBO Dashboard: New Ontology Releases

Overview:
https://obofoundry.github.io/obo-nor.github.io/dashboard/index.html

Detailed results for OntoAvida:
https://obofoundry.github.io/obo-nor.github.io/dashboard/ontovida/dashboard.html

As far as we can see, you have two small issues with how your [version IRI](http://obofoundry.org/principles/fp-004-versioning.html) is formatted, and your [license property](https://github.com/INCATools/ontology-development-kit/blob/master/docs/License.md).

### Next steps:

Once you have completed the above, please let us know by making a comment in this ticket. We will then present your case at the next OBO operations call after that, and select an appropriate reviewer. The reviewer will then provide you with feedback and present your case again to the whole operations committee where the decision will be finalised.

**Comment by miguelfortuna** [comment]
Thanks for taking a look at our ontology. We think we have already solved: 1) the version IRI and 2) the license problem. Is the provisional dashboard updated regularly? We cannot see if the changes have been successful.

Please, let us know if the problems have been finally solved.

Kind regards,
Miguel.

**Comment by matentzn** [comment]
@miguelfortuna thanks!
The Version IRI seems solved, the license issue still is not: file:///Users/matentzn/ws/obo-nor.github.io/dashboard/index.html

The dashboard I think gets tripped by two things:

```
  <terms:license xml:lang="en">OntoAvida by https://fortunalab.org is licensed under CC BY 4.0. You are free to share (copy and redistribute the material in any medium or format) and adapt (remix, transform, and build upon the material) for any purpose, even commercially. for any purpose, even commercially. The licensor cannot revoke these freedoms as long as you follow the license terms. You must give appropriate credit (by using the original ontology IRI for the whole ontology and original term IRIs for individual terms), provide a link to the license, and indicate if any changes were made. You may do so in any reasonable manner, but not in any way that suggests the licensor endorses you or your use.</terms:license>
        <terms:license rdf:datatype="http://www.w3.org/2001/XMLSchema#string">https://creativecommons.org/licenses/by/4.0/</terms:license>
```

1. You shouldn't use `terms:license` to provide a comment about the license - for that, better use `rdfs:comment`
2. The license should be a resource, not a string, and only one such license assertion should exist:

```
<terms:license rdf:resource="https://creativecommons.org/licenses/by/4.0/"/>
```

**Comment by miguelfortuna** [comment]
Done.

Thanks.

**Comment by matentzn** [comment]
Thanks miguel, looks great on the dashboard front. Once you have checked all the checkboxes above, we will proceed with the review.

**Comment by miguelfortuna** [comment]
> @miguelfortuna Thank you very much for your submission! Before we start the review process, we would like to ask you to
> 
>     * check a few boxes in the following checklist, and
> 
>     * ensure that OntoAvida passes the general OBO quality control (OBO Dashboard)
> 
> 
> ### Submission Checklist
> 
> To be considered for inclusion in the OBO Foundry, an ontology must meet certain requirements, as described in [the registration process instructions](http://obofoundry.org/docs/Policy_for_OBO_namespace_and_associated_PURL_requests.html) and the [registration review checklist](https://github.com/OBOFoundry/OBOFoundry.github.io/blob/master/docs/RegistrationChecklist.md). To ensure you are aware of some of its key points, please review the checklist below.
> 
>     * [X]  I have read an understood [the registration process instructions](http://obofoundry.org/docs/Policy_for_OBO_namespace_and_associated_PURL_requests.html) and the [registration checklist](https://github.com/OBOFoundry/OBOFoundry.github.io/blob/master/docs/RegistrationChecklist.md)
> 
>     * [X]  There is no other ontology in the OBO Foundry which would be an appropriate place for my terms. If there was, I have contacted the editors, and we decided in mutual agreement that a separate ontology is more appropriate.
> 
>     * [X]  My ontology has a specific release file with a version IRI and a dc:license annotation, serialised in RDF/XML.
> 
>     * [X]  I understand that term definitions, while not mandatory, are key to understanding the intentions of a term especially when the ontology is used in curation. I made sure that a reasonable majority of terms in my ontology have definitions, using the [IAO:0000115](http://www.ontobee.org/ontology/IAO?iri=http://purl.obolibrary.org/obo/IAO_0000115) property.
> 
>     * [X]  For every term in my ontology, I checked whether another OBO Foundry ontology has one with the same meaning. If so, I re-used that term directly (not by cross-reference, by directly using the IRI).
> 
>     * [X]  For all relationship properties (Object and Data Property) I checked whether [RO](http://www.ontobee.org/ontology/catalog/RO?iri=http://www.w3.org/2002/07/owl%23ObjectProperty) includes an appropriate one. I understand that aligning with RO is an essential part of the overall alignment between OBO ontologies!
> 
>     * [X]  For the selection of appropriate annotation properties, I looked at [OMO](http://www.ontobee.org/ontology/catalog/OMO?iri=http://www.w3.org/2002/07/owl%23AnnotationProperty) first. I understand that aligning ontology metadata and term-level metadata is essential for cross-integration of OBO ontologies.
> 
>     * [X]  If I was not sure about the meaning of any of the checkboxes above, I have consulted with a member of the OBO Foundry for advice.
> 
> 
> ### OBO Dashboard: New Ontology Releases
> 
> Overview:
> https://obofoundry.github.io/obo-nor.github.io/dashboard/index.html
> 
> Detailed results for OntoAvida:
> https://obofoundry.github.io/obo-nor.github.io/dashboard/ontoavida/dashboard.html
> 
> As far as we can see, you have two small issues with how your [version IRI](http://obofoundry.org/principles/fp-004-versioning.html) is formatted, and your [license property](https://github.com/INCATools/ontology-development-kit/blob/master/docs/License.md).
> ### Next steps:
> 
> Once you have completed the above, please let us know by making a comment in this ticket. We will then present your case at the next OBO operations call after that, and select an appropriate reviewer. The reviewer will then provide you with feedback and present your case again to the whole operations committee where the decision will be finalised.

**Comment by miguelfortuna** [comment]
Thanks @matentzn.
Done.

**Comment by matentzn** [comment]
@miguelfortuna thank you, sorry to bother, but would you be able to just "check" the checkboxes I pasted? I will bring your case to the OBO committee today. Thank you!

**Comment by miguelfortuna** [comment]
I can't edit it, or I don't know how to do it. That's why I did it that way. Sorry about it.

**Comment by matentzn** [comment]
Good to know - never thought you may need additional edit rights.. :) Thank you!

**Comment by cmungall** [comment]
Hi @miguelfortuna 

Can you clarify a bit more about the scope of the ontology and whether it's in scope for OBO?

> OntoAvida covers the most widely-used computational approach for studying evolution using digital organisms (i.e., self-replicating computer programs that evolve within a user-defined computational environment).

Do you have some citations on this. E.g. like these

 - https://ieeexplore.ieee.org/abstract/document/6791137
 - https://evolution-outreach.biomedcentral.com/articles/10.1007/s12052-009-0154-z

My view is that this is as much in scope as other ontologies of information entities within OBO but it would be good to have more defense of how this is a scientific enterprise as there is concern that we end up adding ontologies of video games to OBO

Can you also describe more about how the ontology will be used? Would this be used to annotate configuration files for Avida? Is it intended for reasoning about these configurations? Is there some kind of registry of avida programs that this would be used in conjunction with?

Although not a comment on suitability for OBO, my personal view is that OWL may be both overkill and under-utilized for this task. Some people have successfully used OWL for reasoning about configurations about computer programs, determining inconsistent configurations, etc. In order to do this you would need more axioms in your ontology, at least determining what classes are disjoint. But if this really is your goal then there are better languages IMO than OWL for these kinds of constraint satisfaction issues.

The main question we have is why you want to register this with OBO rather than on a broader registry such as BioPortal. By registering with OBO you are indicating that your ontology is intended as part of an interoperable ecosystem with reuse of shared terms and relations. It may be the fact that attempting to do this may actually hinder your goals here, and bioportal may be a better home.

For example, you are subclassing genome and genotype (taken from OGG and NCIT respectively - unfortunately we don't have orthogonality in OBO here but I would recommend using SO).

But regardless of which ontology you take this from, there is problems with this:

![image](https://user-images.githubusercontent.com/50745/132379584-df04c372-4ada-4d10-9ae4-43108f0f26b7.png)

The parent classes are for classic living organisms, not digital simulations or digital entities inspired by these. This isn't anti-silicon bias, you get into all kinds of issues with OWL when you try and subclass a a class whilst simultaneously trying to generalize that class. You are better off using an object property such as simulates or is-model-of or inspired-by

This is also a bit problematic for a variety of reasons:

![image](https://user-images.githubusercontent.com/50745/132380182-075b22ef-c775-4421-b062-40f80d9d2e0c.png)

If you are committed to being a part of OBO then it's not ideal to take a label like "transcribes" which means a certain thing to most OBO users and then use it in the sense of a much more specific or different concept (digital analogs of nucleic acid transcription).

Also the list of subproperties is highly unusual. I can suggest a variety of other modeling patterns for annotating the random number seed, such as reification or modeling the transcription is a process which would be an intermediary between the genome and the transcript, with the seed a plain property of the process.

**Comment by miguelfortuna** [comment]
Thanks for your comments/suggestions. Please, see our responses below your concerns. Note that the "experimental" branch of the gitlab repository (https://gitlab.com/fortunalab/ontoavida/-/tree/experimental) contains the latest version of the ontology. We have incorporated your suggestions into this branch.

---

Can you clarify a bit more about the scope of the ontology and whether it's in scope for OBO?

R: We have added a couple of paragraphs in the README.md file for better describing what Avida is and how well it fits to a collection of biological ontologies like OBO. We will continue improving the README file to provide more information as we update the ontology. 

---

Do you have some citations on this?

R: We have added, at the end of the README.md file, a list of almost 100 scientific publications that have used Avida for shedding light on fundamental evolutionary questions. Digital evolution is an active and growing field of research and our lab is using Avida for the last 10 years.

---

My view is that this is as much in scope as other ontologies of information entities within OBO but it would be good to have more defense of how this is a scientific enterprise as there is concern that we end up adding ontologies of video games to OBO

R: Avida is a solid computational approach for studying evolution. It has been used in very influential papers published in top interdisciplinary scientific journals like Science and Nature.

---

Can you also describe more about how the ontology will be used? Would this be used to annotate configuration files for Avida? Is it intended for reasoning about these configurations? Is there some kind of registry of avida programs that this would be used in conjunction with?

R: We have explicitly mentioned in the README.md file that, besides formalizing a common vocabulary for performing in silico evolution experiments using Avida, the ontology goals are: 1) make inferences based on certain rules and constraints (also inferring object properties like "computes" by using property chain axioms), 2) facilitate the reproducibility of the in silico evolution experiments reported in the scientific literature, and 3) trace the provenance of the data stored in AvidaDB, a RDF database on digital organisms that comprises more than a million genomes (available soon).

---

Although not a comment on suitability for OBO, my personal view is that OWL may be both overkill and under-utilized for this task. Some people have successfully used OWL for reasoning about configurations about computer programs, determining inconsistent configurations, etc. In order to do this you would need more axioms in your ontology, at least determining what classes are disjoint. But if this really is your goal then there are better languages IMO than OWL for these kinds of constraint satisfaction issues.

R: OntoAvida is not intended for reasoning about configuration programs, but for reasoning about how evolution unfolds (e.g., how genetic information is encoded, translated into function, and influenced by the environment).

---

The main question we have is why you want to register this with OBO rather than on a broader registry such as BioPortal. By registering with OBO you are indicating that your ontology is intended as part of an interoperable ecosystem with reuse of shared terms and relations. It may be the fact that attempting to do this may actually hinder your goals here, and bioportal may be a better home.

R: That's exactly why we want to register OntoAvida with OBO: we want OntoAvida to be an ontology that is part of an interoperable ecosystem with reuse of shared terms and relations. We want to keep things as simple as possible.

---

For example, you are subclassing genome and genotype (taken from OGG and NCIT respectively - unfortunately we don't have orthogonality in OBO here but I would recommend using SO).

R: We missed the SO ontology! We have now used it for the classes "genome" and "genotype".

---

If you are committed to being a part of OBO then it's not ideal to take a label like "transcribes" which means a certain thing to most OBO users and then use it in the sense of a much more specific or different concept (digital analogs of nucleic acid transcription).

R: We have replaced the object property "transcribes" by "executes". A digital organism executes the instructions it harbors in its genome (some of them once or more times), and that list of executed instructions is what we call "digital organism transcriptome".

---

The parent classes are for classic living organisms, not digital simulations or digital entities inspired by these. This isn't anti-silicon bias, you get into all kinds of issues with OWL when you try and subclass a a class whilst simultaneously trying to generalize that class. You are better off using an object property such as simulates or is-model-of or inspired-by

R: We think the class "genome" should not be restricted to a chemistry-based information storing device. In that sense, the genome of a digital organism should be a subclass of "genome". But we see your point and will think about it.

---

Also the list of subproperties is highly unusual. I can suggest a variety of other modeling patterns for annotating the random number seed, such as reification or modeling the transcription is a process which would be an intermediary between the genome and the transcript, with the seed a plain property of the process.

R: We have spent hours thinking the best way to do it. Unfortunately, we couldn't find any other way. In Avida, digital organisms can show phenotypic plasticity. Therefore, the same genome can encode distinct phenotypes depending on the environment (which is determined by the seed used for starting the pseudo-random number generator). Different genomes can encode the same phenotype in the same environment (seed) as well. So, we didn't find any better way to keep the triples: 

genome_1-> encodes -> phenotype_1 (at seed = 1)

genome_1-> encodes -> phenotype_1 (at seed = 2)

genome_1-> encodes -> phenotype_2 (at seed = 3)

genome_2-> encodes -> phenotype_1 (at seed = 1)

---

Please, let us know any other suggestion/comment you might have.

Thanks.

Kind regards,
Miguel.

**Comment by cmungall** [comment]
Thank you for your attentive response. Apologies for the delay. In the interests of time I just want to hone in on one response to illustrate certain issues:

> We think the class "genome" should not be restricted to a chemistry-based information storing device. In that sense, the genome of a digital organism should be a subclass of "genome". But we see your point and will think about it.

This is a reasonable viewpoint - but the broader issue here is that it doesn't work in OBO if people start repurposing existing classes, and broadening them beyond what is implicitly or explicitly intended by the provider of that class. This will lead to all kinds of problems and inconsistencies downstream. In this case it would be reasonable for you to make an issue on the SO tracker explaining your position and requesting a generalization. Even if unsuccessful this is part of the general engagement process in making a collection of ontologies that work together in a reasonable way.

Note that for many reasons, I think the consensus will be that genotypes are restricted to exclude digital genotypes. However, it would be perfectly reasonable to make a class "digital genotype" with some relationship such as simulation-of or analog-of to the SO class. Now it may be annoying for your users to see a primary label "digital genotype" when to them "genotype" is already digital. But that's one of the compromises of being part of OBO. Is this process still of interest to you given some of these compromises?

You may also want to explore IAO which is for information entities. I would suggest many of your terms belong there. A caveat is that IAO is not well maintained and there is not much guidance on how things like simulations are represented in OBO

**Comment by cmungall** [comment]
Normally we would require ontologies to "better fit" within OBO before adding. For example, a new anatomy ontology for a metazoan should fit within Uberon.

However, even though the right place for your ontology is as subclasses of IAO classes (or even folded into IAO) we do not have good documentation or patterns on how to represent digital simulations in IAO. So I think we should go ahead and add your ontology, so long as you are willing to engage with other ontologies in OBO to ensure consistent placement of terms. the OBO operations committee concurs with this

**Comment by miguelfortuna** [comment]
Dear Chris,

Thank you very much for your willingness to add our ontology. We are exploring IAO to incorporate your suggestions and uploading an updated version of OntoAvida in the master branch of the GitLab repository. It will be done in a few days. Our computer technician, Raul Ortega, will be actively involved in the OBO community. Is there anything else we need to do or any additional information you need?

**Comment by miguelfortuna** [comment]
New term request to Sequence Ontology:
https://github.com/The-Sequence-Ontology/SO-Ontologies/issues/559

**Comment by miguelfortuna** [comment]
Hi @cmungall - the first release of OntoAvida is already in the master branch of GitLab (owl and obo files):
https://gitlab.com/fortunalab/ontoavida

The GitLab username of our computer technician is:
@raul.ortega

Thanks for your feedback.

**Comment by miguelfortuna** [comment]
We have also uploaded a HTML version of the ontology at:
https://owl.fortunalab.org/ontoavida
It can also be found at the development branch of our GitLab repository:
https://gitlab.com/fortunalab/ontoavida/-/blob/development/ontoavida_LODE.html

**Comment by cmungall** [comment]
Thank you for the updated version. I have some minor comments that I will add to your gitlab issue tracker, but this addresses the main concerns.

**Comment by cmungall** [comment]
We decided on 2021-11-30 call to accept this ontology, welcome! 

Next steps are to add the metadata for this ontology - are you OK to make the PR for this or would you like assistance with this?

**Comment by matentzn** [comment]
This is the metadata file:

```
- id: ontoavida
      title: "OntoAvida: ontology for Avida digital evolution platform"
      contact:
        email: fortuna@ebd.csic.es
        label: Miguel A. Fortuna
      description: "OntoAvida develops an integrated vocabulary for the description of the most widely-used computational approach for studying evolution using digital organisms (i.e., self-replicating computer programs that evolve within a user-defined computational environment)."
      domain: digital evolution, a branch of Artificial Life.
      homepage: https://gitlab.com/fortunalab/ontoavida
      products:
        - id: ontoavida.owl
        - id: ontoavida.obo
      dependencies:
        - id: fbcv
        - id: gsso
        - id: ogg
        - id: stato
        - id: ro
        - id: ncit
      tracker: https://gitlab.com/fortunalab/ontoavida/-/issues
      license:
        url: https://creativecommons.org/licenses/by/4.0/
        label: CC BY 4.0
```

Please use this to make a pull request similar to this one:
https://github.com/OBOFoundry/OBOFoundry.github.io/pull/1645/files

Then make a PR to create a file like this one:
https://github.com/OBOFoundry/purl.obolibrary.org/blob/master/config/aism.yml

in the PURL repo.

**Comment by miguelfortuna** [comment]
Thanks a lot! @cmungall 

We have done the pull requests:
https://github.com/OBOFoundry/purl.obolibrary.org/pull/807
https://github.com/OBOFoundry/OBOFoundry.github.io/pull/1673

Please, let us know if everything works as expected.

**Comment by miguelfortuna** [comment]
Thanks a lot!

**Comment by matentzn** [comment]
This is done!
